### PR TITLE
Emma development

### PR DESCRIPTION
**Pull request by emmapaczkowski**
I have added the Google Maps API to my portfolio as well as the comment feature.

### CODE REVIEW

**Comment by saileshsimhadri** [comment]
Jasmine, I know that there is a bunch of weird formatting changes. Emma and I chatted offline, and there was something weird that happened during some of the merges. We're hoping that after this pr is merged, it should be back to normal.

File: walkthroughs/week-3-server/step-3-json-walkthrough.md [hunk 1]
```diff
@@ -60,9 +59,9 @@ Here is an array that contains three objects:
 
 ```json
 [
-  {"name": "Ada", "birthYear": 1815},
-  {"name": "Grace", "birthYear": 1906},
-  {"name": "Alan", "birthYear": 1912}
+  { "name": "Ada", "birthYear": 1815 },
```
Comment (saileshsimhadri) [inline walkthroughs/week-3-server/step-3-json-walkthrough.md:62]: For here an below, undo the extra spaces added?

File: walkthroughs/week-4-libraries/maps/examples/hello-world/src/main/webapp/index.html [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-  <script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY"></script>
-  <script src="script.js"></script>
-  <link rel="stylesheet" href="style.css">
-</head>
-<body onload="createMap();">
-  <h1>Hello Google Maps</h1>
-  <div id="map"></div>
-  <p>This page shows a Google Map!</p>
-</body>
+  <head>
```
Comment (saileshsimhadri) [inline walkthroughs/week-4-libraries/maps/examples/hello-world/src/main/webapp/index.html:3]: For all these files, remove these extra spaces.

**Comment by saileshsimhadri** [comment]
There's a lot of formatting changes that you made. I realize that you were having issues, and its fine for now. Just make sure to make them a separate pull request next time. 

But can you double check that these are intended. Many seem like they were accidental and things you don't want.

File: walkthroughs/week-3-server/step-3-json-walkthrough.md (hunk 1 above)
Comment (emmapaczkowski) [reply walkthroughs/week-3-server/step-3-json-walkthrough.md:62]: That is a file I was given by them to look at. I did not change it.

File: walkthroughs/week-4-libraries/maps/examples/hello-world/src/main/webapp/index.html (hunk 2 above)
Comment (emmapaczkowski) [reply walkthroughs/week-4-libraries/maps/examples/hello-world/src/main/webapp/index.html:3]: That is a file I was given by them to look at. I did not change it.

File: portfolio/src/main/java/com/google/sps/classes/Comment.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private Date time;
+
+    public Comment(String message, String name, Date time) {
+        this.message = message;
+        this.name = name;
+        this.time = time;
+    }
+
+    public String getMessage() {
+        return this.message;
```
[the remark this answers is not in the document]
Comment (emmapaczkowski) [reply portfolio/src/main/java/com/google/sps/classes/Comment.java:17]: done

[the remark this answers is not in the document]
Comment (emmapaczkowski) [reply portfolio/src/main/java/com/google/sps/classes/Comment.java:17]: done

File: walkthroughs/week-3-server/step-3-json-walkthrough.md (hunk 1 above)
Comment (emmapaczkowski) [reply walkthroughs/week-3-server/step-3-json-walkthrough.md:62]: That is a file I was given by them to look at. I did not change it.

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(newComment);
+
+    response.sendRedirect("https://8080-03adb9af-0470-4f43-b5b3-40607f387072.us-west1.cloudshell.dev/?authuser=0");
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    value = (value == null) ? defaultValue : value;
```
[the remark this answers is not in the document]
Comment (emmapaczkowski) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:88]: done

File: walkthroughs/week-4-libraries/maps/examples/hello-world/src/main/webapp/index.html (hunk 2 above)
Comment (saileshsimhadri) [reply walkthroughs/week-4-libraries/maps/examples/hello-world/src/main/webapp/index.html:3]: Hmm ok. The github diff shows that you did, but maybe its git being weird?

**Comment by saileshsimhadri** [comment]
LGTM. I'll let Jasmine approve.

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      String name = (String) entity.getProperty("name");
+      String message = (String) entity.getProperty("message");
+      Date time = (Date) entity.getProperty("timestamp");
+
+      commentList.add(new Comment(name, message, time));
+    }
+
+    int numCommentsRequested = getNumComments(request);
+
+     if (commentList.size() > numCommentsRequested){
```
[the remark this answers is not in the document]
Comment (wjiamin) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:44]: Sorry what was the conclusion for this? Is this case possible?

[the remark this answers is not in the document]
Comment (emmapaczkowski) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:44]: The way I wrote the code was that regardless of how many comments were requested, you would iterate through the entity comments that you have and add them to the commentList array list (line 33-40).  In the case that you have requested more comments than you have which is what I believe you were asking, then you would show all the comments that you have.  In the case you are requesting to see a smaller number of comments than you have, then, the if statement on line 44-46 creates a sub list only showing the number of comments you requested.

[the remark this answers is not in the document]
Comment (wjiamin) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:44]: Thanks for the explanation, that makes sense.

**Pull request merged**